### PR TITLE
Refactor RecordingActivity

### PR DESCRIPTION
**Pull request by cmsigrist**
Refactor the RecordignActivity so that it can be called from any Activity

### CODE REVIEW

**Review by Xelowak — APPROVED** [review]
LGTM ! :)

File: app/src/androidTest/java/ch/epfl/sdp/blindly/main_screen/ProfilePageTest.kt [hunk 1]
```diff
@@ -143,29 +144,6 @@ class ProfilePageTest {
...
-
-        onView(withId(R.id.record_button)).perform(click())
-        val fragmentManager = fragment.childFragmentManager
-        val audioFragment = fragmentManager.findFragmentById(R.id.fragment_audio_container_view)
-        assert(audioFragment == null)
-    }
-
-     */
-
     private fun getAudioPlayerFragment(): Fragment? {
```
Comment (cgntnr) [inline app/src/androidTest/java/ch/epfl/sdp/blindly/main_screen/ProfilePageTest.kt:147]: Why is this private function hanging out here all alone :D ? It seems other tests are deleted and no one is using this function.

File: app/src/main/java/ch/epfl/sdp/blindly/main_screen/audio_player/AudioPlayerFragment.kt [hunk 2]
```diff
@@ -1,122 +1,118 @@
...
+import ch.epfl.sdp.blindly.user.UserHelper
+import java.io.File
+import javax.inject.Inject
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [AudioPlayerFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+
```
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/main_screen/audio_player/AudioPlayerFragment.kt:30]: I think it's Github's doing but it seems like all lines are removed and added as git diff, which made it hard to find the changes.

**Review by cgntnr — APPROVED** [review]
Good amount of code in PR. It's good that some non-null assertions are replaced to avoid problems. The PR changed its purpose midway I believe so I'm not sure @Xelowak 's first review is relavant anymore, because the test added is removed and other changes/refactoring are done after he reviewed.

File: app/src/main/java/ch/epfl/sdp/blindly/main_screen/audio_player/AudioPlayerFragment.kt (hunk 2 above)
Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/main_screen/audio_player/AudioPlayerFragment.kt:30]: Yep it's a bit weird sorry :/

**Pull request merged**